lib/active_publisher/async.rb

```diff
@@ -0,0 +1,29 @@
+module ActivePublisher
+  # Publish a message asynchronously to RabbitMQ.
+  #
+  # Asynchronous is designed to do two things:
+  # 1. Introduce the idea of a durable retry should the RabbitMQ connection disconnect.
+  # 2. Provide a higher-level pattern for fire-and-forget publishing.
+  #
+  # @param [String] route The routing key to use for this message.
+  # @param [String] payload The message you are sending. Should already be encoded as a string.
+  # @param [String] exchange The exchange you want to publish to.
+  # @param [Hash] options hash to set message parameters (e.g. headers).
+  def self.publish_async(route, payload, exchange_name, options = {})
+    ::ActivePublisher::Async.publisher_adapter.publish(route, payload, exchange_name, options)
+  end
+
+  module Async
+    def self.publisher_adapter
+      @publisher_adapter ||= case ::ActivePublisher.configuration.async_publisher
```

Comment:
@mmmries should we leave it a little abstracted for the client side so they don't have to know exactly which class to instantiate in the gem and instead just pass in the string like they do in the configuration right now? I'm thinking something like this:

```
module Async
  def self.publisher_adapter
    @publisher_adapter
  end

  def self.publisher_adapter=(adapter)
    @publisher_adapter = case adapter
                         when /memory/i then
                           require "active_publisher/async/in_memory_adapter"
                           ::ActivePublisher::Async::InMemoryAdapter.new
                         when /redis/i then
                           fail "Not yet implemented"
                         else
                           fail "Unknown adapter '#{::ActivePublisher.configuration.async_publisher}' provided"
                         end
  end
end
```